### DISCUSSION TITLE
Mirror to Codeberg

### DISCUSSION THREAD

**Discussion by foxxy432** [FreshRSS/FreshRSS#5335] [Ideas]
### Summary

We should mirror our repos to Codeberg because:

Codeberg is libre, free, open-source, privacy-friendly, non-profit
GitHub is privacy-invasive, proprietary
People on Codeberg like me can contribute to the project
GitHub can take down your repos

FAQ

Q: - GitHub Actions -- this is a huge time saver for me
- Forcing all contributors to modify their current setup and move to another hosting -- we may just lose contributors in the end

A: As I said, we only just mirror. The repo will be available and contributable on both sites.

Q: - Transferring issues -- is this possible? Issues here represent a technical knowledge base we can't live without
- Transferring wiki and all edit history -- possible?

A: Yes, it is possible.
a. Example for issues:

Migration to independent platform for provision of source code andersju/webbkoll#35
https://codeberg.org/dataskydd.net/webbkoll/issues/35
[UI/UX] Simplification of the user interface for non-technical people andersju/webbkoll#33
https://codeberg.org/dataskydd.net/webbkoll/issues/33

Framed answers? - Locations zyachel/quetre#7
https://codeberg.org/zyachel/quetre/issues/7

b. Example for commits:

https://codeberg.org/dataskydd.net/webbkoll/commit/6ddfbc6c71ab92f754252dd4904cf49fb19672a6
andersju/webbkoll@6ddfbc6

c. Example for wiki:

https://github.com/Freeyourgadget/Gadgetbridge/wiki/Home/_history?page=2
https://codeberg.org/Freeyourgadget/Gadgetbridge/wiki/Home?action=_revision

Q: Despite having mirror/s, the project will still continue to be operated from GitHub for development, so having yet another mirror doesn't solve anything.

A: You can create issues, contribute, view the repos on there without going to GitHub! Example:

https://github.com/libredirect/libredirect
https://codeberg.org/LibRedirect/libredirect
https://github.com/zyachel/quetre
https://codeberg.org/zyachel/quetre

Q: How does Codeberg mirror comments from GitHub?

A: It uses Access Token.

Q: My repos are very bandwidth-intensive projects, and I have my doubts as to whether the owner(s) of Codeberg would be prepared for such a massive bandwidth rate increase. I've also had individual projects be disabled on small sites before due to taking too much bandwidth.

A: Oh.. I think you should directly ask https://docs.codeberg.org/contact/ about if they are ready for massive bandwidth rate increase.

### What benefits does this feature provide?

Codeberg is libre, free, open-source, privacy-friendly, non-profit
GitHub is privacy-invasive, proprietary
People on Codeberg like me can contribute to the project
GitHub can take down your repos

**Comment by Alkarex** [comment]
Hello,
There are multiple places where we could also be hosted or mirrored. We are for instance quite close to the [Framasoft](https://framasoft.org/en/) community, who is using https://framagit.org

Personally, I keep an eye on git federation efforts (cf. e.g. https://github.com/go-gitea/gitea/issues/18240 ), which would allow a true distribution.

I am not familiar with Codeberg, so I would have a few more questions to clear, such as:

* What minimal access token permissions would be needed to be meaningful? It should not be permissions higher than what a normal logged-in external GitHub user has.

* I can see that the examples https://codeberg.org/zyachel/quetre/issues and https://github.com/zyachel/quetre/issues have different content. That probably means that someone needs to also monitor the activity there (yet another channel), with risks of duplicates, right?
    * A read-only copy could maybe be an option, if it make sense at all.

* Risks of resource waste if it is not actually used by enough people.

**Reply by Frenzie** [reply]
As a sidenote there's also https://sr.ht/

I use Gitea daily; it's very nice. This Codeberg place is clearly some kind of hosted Gitea btw.

**Comment by math-GH** [comment]
I do not see any benefits. Somewhere we had already the discussion to move to a Gitlab server.

Codeberg has the same issue like Github: If they will close the server they will close the access. Codeberg is so unknown and small, that no one will take care in case of shut down.

Github is very beginner friendly. That was the reason why I started to contribute here.

**Comment by math-GH** [comment]
One more comment regarding "Codeberg" and its "Codeberg e.V.":
(Background: I am from Germany)
Codeberg e.V. is not transparent and not very international.
Membership only for Germans? https://join.codeberg.org/
Who are the board members of the "e.V." (e.V. is the German abbreviation for "registered association")? And any information about the articles of association (German: Satzung)?

To use a platform means to trust them. Why should someone trust Codeberg? Why more than GitHub?
